android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUITimePicker.java

```diff
@@ -160,15 +174,36 @@ public void setValue(long value, boolean suppressEvent)
 	@Override
 	public void onTimeChanged(TimePicker view, int hourOfDay, int minute)
 	{
+		detectTimeChange();
+	}
+	
+	private void detectTimeChange() {
 		Calendar calendar = Calendar.getInstance();
-		calendar.set(Calendar.HOUR_OF_DAY, hourOfDay);
-		calendar.set(Calendar.MINUTE, minute);
-		if (!suppressChangeEvent) {
+		TimePicker picker = (TimePicker) getNativeView();
+		boolean hasChanged;
+		if (Build.VERSION.SDK_INT >= 23) {
+			hasChanged = lastSelectedHour != picker.getHour() || lastSelectMin != picker.getMinute();
```

Comment:
I don't agree. For one-off API calls like this, we should use an inner class as I've shown above. It keeps it simple and it keeps the code together in one place... making it easier to maintain and understand. And the ApiLevel23 class name makes it clear to the developer that it should only be accessed by that API level or higher. Also, by making it an inner class, it effectively becomes a private API that module developers cannot use (this is intentional; we're hding the complexity from a public API standpoint).

Moving higher API level calls to another class is only useful if it's needed by multiple classes. Let's avoid this unless needed, because once we create a public class in an open source project, it's set in stone. We'll have to maintain it and provide tech-support for it.